src/qibo/transpiler/unroller.py

```diff
@@ -142,17 +147,22 @@ def assert_decomposition(
 def translate_gate(
     gate,
     native_gates: NativeGates,
+    backend=None,
 ):
     """Maps gates to a hardware-native implementation.
 
     Args:
         gate (:class:`qibo.gates.abstract.Gate`): gate to be decomposed.
         native_gates (:class:`qibo.transpiler.unroller.NativeGates`): native gates
             to use in the decomposition.
+        backend (:class:`qibo.backends.Backend`): backend to use for gate matrix.
 
     Returns:
         list: List of native gates that decompose the input gate.
     """
+    if backend is None:
+        backend = _check_backend(backend)
```

Comment:
`_check_backend` already handles the `None` case

```suggestion
    backend = _check_backend(backend)
```